### PR TITLE
Simplifies config for (not) using encryption.

### PR DESCRIPTION
**Pull request by technige**

### CODE REVIEW

File: driver/src/main/java/org/neo4j/driver/v1/Config.java [hunk 1]
```diff
@@ -156,9 +156,18 @@ public int connectionTimeoutMillis()
...
     public EncryptionLevel encryptionLevel()
     {
-        return encryptionLevel;
+        return encrypted ? EncryptionLevel.REQUIRED : EncryptionLevel.NONE;
+    }
+
+    /**
+     * @return indicator for encrypted communication.
+     */
+    public boolean encrypted()
```
Comment (lutovich) [inline driver/src/main/java/org/neo4j/driver/v1/Config.java:168]: What do you think about `isEncrypted` name for this method?

File: driver/src/main/java/org/neo4j/driver/v1/Config.java [hunk 2]
```diff
@@ -328,9 +337,30 @@ public ConfigBuilder withConnectionLivenessCheckTimeout( long value, TimeUnit un
          * @param level the TLS level to use
          * @return this builder
          */
+        @Deprecated
```
Comment (lutovich) [inline driver/src/main/java/org/neo4j/driver/v1/Config.java:340]: `EncryptionLevel` enum should probably also be deprecated

**Review by lutovich — COMMENTED** [review]
@technige PR looks good, couple small questions

**Comment by zhenlineo** [comment]
Interesting we already have `public boolean logLeakedSessions()`. 
So maybe we should keep `public boolean encrypted()` instead? CC @lutovich

**Comment by lutovich** [comment]
@zhenlineo yeah, let's then keep `#encrypted()`

**Pull request merged**